### PR TITLE
[youporn] fix url metadata detection (width and bits) to allow best v…

### PR DESCRIPTION
**Pull request by Kasperdilday**
The slash in the end seems to have gone, metadata format has changed. This fixes best file selection.

[x] I am not the original author of this code but it is in public domain or released under Unlicense (provide reliable evidence)

What is the purpose of your pull request?

[x] Bug fix

Please see more details below.

### CODE REVIEW

**Comment by dstftw** [comment]
Provide concrete examples of what it's supposed to fix.

**Comment by Kasperdilday** [comment]
Oh, yes, sorry. I see the problem. Last second I switched to the example video from the file to prepare the URL example and it noticed just now that it is different, as in working with the "old" code.

There are a few examples where the `videoUrl` has a different format and adapted the regex so that it works for both versions.

`/videos\/201703\/11\/109285532\/720P_4000K_109285532.mp4?rate=248k&burst=1400k&validfrom=1553107800&validto=1553122200&hash=NzBS4CUWB2RpgA9thDRS0Ouw5PM%3D`

Interesting is the `.mp4` where there is a `/` in other URLs.

I added the example to the list in the code.

Speaking of output difference:

```
without fix:
youtube-dl --list-formats https://www.youporn.com/watch/13922959/femdom-principal/  
[YouPorn] femdom-principal: Downloading webpage
[info] Available formats for 13922959:
format code  extension  resolution note
0            mp4        unknown    
1            mp4        unknown    
2            mp4        unknown    (best)

with fix:
youtube-dl --list-formats https://www.youporn.com/watch/13922959/femdom-principal/
[YouPorn] femdom-principal: Downloading webpage
[info] Available formats for 13922959:
format code  extension  resolution note
240p-400k    mp4        240p        400k 
480p-2000k   mp4        480p       2000k 
720p-4000k   mp4        720p       4000k  (best)

regression test:

youtube-dl --list-formats https://www.youporn.com/watch/505835/sex-ed-is-it-safe-to-masturbate-daily/
[YouPorn] sex-ed-is-it-safe-to-masturbate-daily: Downloading webpage
[info] Available formats for 505835:
format code   extension  resolution note
180p-150k     3gp        180p        150k 
240p-240k-0   mp4        240p        240k 
240p-240k-1   mp4        240p        240k 
480p-750k-0   mp4        480p        750k 
480p-750k-1   mp4        480p        750k 
720p-1500k-0  mp4        720p       1500k 
720p-1500k-1  mp4        720p       1500k  (best)
```

**Comment by dstftw** [comment]
Add a test.

File: youtube_dl/extractor/youporn.py [hunk 1]
```diff
@@ -119,8 +119,9 @@ def _real_extract(self, url):
             # Video URL's path looks like this:
             #  /201012/17/505835/720p_1500k_505835/YouPorn%20-%20Sex%20Ed%20Is%20It%20Safe%20To%20Masturbate%20Daily.mp4
             #  /201012/17/505835/vl_240p_240k_505835/YouPorn%20-%20Sex%20Ed%20Is%20It%20Safe%20To%20Masturbate%20Daily.mp4
+            #  /videos/201703/11/109285532/720P_4000K_109285532.mp4?rate=248k&burst=1400k&validfrom=1553107800&validto=1553122200&hash=NzBS4CUWB2RpgA9thDRS0Ouw5PM%3D
             # We will benefit from it by extracting some metadata
-            mobj = re.search(r'(?P<height>\d{3,4})[pP]_(?P<bitrate>\d+)[kK]_\d+/', video_url)
+            mobj = re.search(r'(?P<height>\d{3,4})[pP]_(?P<bitrate>\d+)[kK]_\d+(/|\.\w{3}\?)', video_url)
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: Last part should be removed instead.

Comment (Kasperdilday) [reply youtube_dl/extractor/youporn.py]: Sure

**Comment by Kasperdilday** [comment]
Added a test and fixed the existing ones - anything more I can do to make this acceptable?

File: youtube_dl/extractor/youporn.py [hunk 2]
```diff
@@ -145,9 +160,8 @@ def _real_extract(self, url):
             r'(?s)<div[^>]+class=["\']submitByLink["\'][^>]*>(.+?)</div>',
             webpage, 'uploader', fatal=False)
         upload_date = unified_strdate(self._html_search_regex(
-            [r'Date\s+[Aa]dded:\s*<span>([^<]+)',
-             r'(?s)<div[^>]+class=["\']videoInfo(?:Date|Time)["\'][^>]*>(.+?)</div>'],
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: Do not remove old patterns.

File: youtube_dl/extractor/youporn.py [hunk 3]
```diff
@@ -158,9 +172,7 @@ def _real_extract(self, url):
         view_count = str_to_int(self._search_regex(
             r'(?s)<div[^>]+class=(["\']).*?\bvideoInfoViews\b.*?\1[^>]*>.*?(?P<count>[\d,.]+)<',
             webpage, 'view count', fatal=False, group='count'))
-        comment_count = str_to_int(self._search_regex(
-            r'>All [Cc]omments? \(([\d,.]+)\)',
-            webpage, 'comment count', fatal=False))
+        comment_count = len(re.findall(r'<div[^>]+class=([\"\']).*?videoComment\b.*?\1', webpage))
```
Comment (dstftw) [inline youtube_dl/extractor/youporn.py]: Incorrect.

**Pull request closed without merging**